src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -54,19 +57,56 @@ void ColumnBetweenTableScanImpl::_scan_generic_segment(const BaseSegment& segmen
 
     auto typed_left_value = type_cast_variant<ColumnDataType>(_left_value);
     auto typed_right_value = type_cast_variant<ColumnDataType>(_right_value);
-    auto comparator = [typed_left_value, typed_right_value](const auto& position) {
-      return position.value() >= typed_left_value && position.value() <= typed_right_value;
-    };
 
-    _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
+    if (_left_inclusive) {
+      if (_right_inclusive) {
+        auto comparator = [typed_left_value, typed_right_value](const auto& position) {
+          return position.value() >= typed_left_value && position.value() <= typed_right_value;
+        };
+        _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
+      } else {
```

Comment:
I think having four ifs `_left_inclusive && _right_inclusive`, `_left_inclusive && !...` would be easier to read.

Reply:
wouldn't it be less performant?

Reply:
I have no idea how the compiler translates those two versions, but I wouldn't be surprised if the result doesn't look anything like either. Given that this is control-level code, not data-level code, readability seems to be of higher performance than a minimal performance improvement that will probably not be discernible from noise.